### PR TITLE
EXPOSERS: Delete User

### LINKED ISSUE #116: EXPOSERS: Delete User

### PR DESCRIPTION
**Pull request by jasurbekyusuf**
Closes #116

### CODE REVIEW

File: Tarteeb.Api/Controllers/UsersController.cs [hunk 1]
```diff
@@ -100,5 +100,42 @@ public async ValueTask<ActionResult<User>> GetUserByIdAsync(Guid Id)
                 return InternalServerError(userServiceException.InnerException);
             }
         }
+        [HttpDelete("{userId}")]
```
Comment (Javlonbek-dev) [inline Tarteeb.Api/Controllers/UsersController.cs:103]: Add Line

File: Tarteeb.Api/Controllers/UsersController.cs [hunk 2]
```diff
@@ -100,5 +100,43 @@ public async ValueTask<ActionResult<User>> GetUserByIdAsync(Guid Id)
                 return InternalServerError(userServiceException.InnerException);
             }
         }
+
+        [HttpDelete("{userId}")]
+        public async ValueTask<ActionResult<User>> DeleteUserByIdAsync(Guid usertId)
```
Comment (jamshidbek001) [inline Tarteeb.Api/Controllers/UsersController.cs:105]: change parameter to userId

**Review by jamshidbek001 — CHANGES_REQUESTED** [review]
check parameter

**Review by jamshidbek001 — APPROVED** [review]

**Review by Javlonbek-dev — APPROVED** [review]

File: Tarteeb.Api/Controllers/UsersController.cs [hunk 3]
```diff
@@ -100,5 +100,43 @@ public async ValueTask<ActionResult<User>> GetUserByIdAsync(Guid Id)
...
+            {
+                return Locked(userDependencyValidationException.InnerException);
+            }
+            catch (UserDependencyValidationException userDependencyValidationException)
+            {
+                return BadRequest(userDependencyValidationException);
+            }
+            catch (UserDependencyException userDependencyException)
+            {
+                return InternalServerError(userDependencyException);
```
Comment (ElbekDeveloper) [inline Tarteeb.Api/Controllers/UsersController.cs:134]: Nega inner exception qaytarilmayapti?

File: Tarteeb.Api/Controllers/UsersController.cs [hunk 4]
```diff
@@ -100,5 +100,43 @@ public async ValueTask<ActionResult<User>> GetUserByIdAsync(Guid Id)
...
+            }
+            catch (UserServiceException userServiceException)
+            {
+                return InternalServerError(userServiceException);
```
Comment (ElbekDeveloper) [inline Tarteeb.Api/Controllers/UsersController.cs:138]: Bunda ham

File: Tarteeb.Api/Controllers/UsersController.cs [hunk 5]
```diff
@@ -100,5 +100,43 @@ public async ValueTask<ActionResult<User>> GetUserByIdAsync(Guid Id)
...
+                return BadRequest(userValidationException.InnerException);
+            }
+            catch (UserDependencyValidationException userDependencyValidationException)
+                when (userDependencyValidationException.InnerException is LockedUserException)
+            {
+                return Locked(userDependencyValidationException.InnerException);
+            }
+            catch (UserDependencyValidationException userDependencyValidationException)
+            {
+                return BadRequest(userDependencyValidationException);
```
Comment (ElbekDeveloper) [inline Tarteeb.Api/Controllers/UsersController.cs:130]: Bunda ham

**Review by ElbekDeveloper — CHANGES_REQUESTED** [review]

**Review by ElbekDeveloper — APPROVED** [review]

**Pull request merged**